engine/internal/runtime/manager.go

```diff
@@ -169,27 +208,31 @@ func (m *Manager) GetLLMAddress(modelID string) (string, error) {
 	return "", fmt.Errorf("runtime for model %q is not ready", modelID)
 }
 
-// ListSyncedModelIDs returns the list of models that are synced.
-func (m *Manager) ListSyncedModelIDs() []string {
+// ListSyncedModels returns the list of models that are synced.
+func (m *Manager) ListSyncedModels() []*ModelRuntimeInfo {
 	return m.listModels(true)
 }
 
 // ListInProgressModels returns the list of models that are in progress.
-func (m *Manager) ListInProgressModels() []string {
+func (m *Manager) ListInProgressModels() []*ModelRuntimeInfo {
 	return m.listModels(false)
 }
 
-func (m *Manager) listModels(ready bool) []string {
+func (m *Manager) listModels(ready bool) []*ModelRuntimeInfo {
```

Comment:
Just wanted to make sure `ModelRuntimeInfo` is immutable since we return them to a caller without copying.